ibm_db2/tests/test_integration.py

```diff
@@ -129,3 +130,19 @@ def test_metadata(instance, datadog_agent, dd_run_check):
     }
 
     datadog_agent.assert_metadata(CHECK_ID, version_metadata)
+
+
+@pytest.mark.usefixtures('dd_environment')
+def test_disconnection(aggregator, instance, dd_run_check):
+    instance["connection_timeout"] = 1
+    check = IbmDb2Check('ibm_db2', {}, [instance])
+
+    # Disconnect the database
+    run_command('docker exec ibm_db2 su - db2inst1 -c "db2stop force"', check=True)
+    dd_run_check(check)
+    aggregator.assert_service_check(check.SERVICE_CHECK_CONNECT, check.CRITICAL)
+
+    # # Reconnect the database
+    run_command('docker exec ibm_db2 su - db2inst1 -c "db2start"', check=True)
+    check.check(instance)
+    aggregator.assert_service_check(check.SERVICE_CHECK_CONNECT, check.OK)
```

Comment:
I'm a bit wary of this. Can we test this without manipulating the environment? Wouldn't setting the database / host we connect to to something that doesn't exist be enough to test that the connection timeout is being respected?

Reply:
I was trying to be as close as possible to what the customer expirence but if you have a beter idea let me know. i'm not very familiar with the testing framework.

Reply:
See https://github.com/DataDog/integrations-core/pull/14110#discussion_r1331599916.